lib/cequel/record/lazy_record_collection.rb

```diff
@@ -77,12 +74,21 @@ def assert_fully_specified!
 
       attr_reader :record_set
 
-      def_delegators :record_set, :key_columns, :scoped_key_values
-      private :key_columns, :scoped_key_values
-
       def key_attributes_for_each_row
         map { |record| record.key_attributes }
       end
+
+      def record_set_delegated_methods
+        %i[table connection key_columns scoped_key_values]
+      end
+
+      def method_missing(m, *args, &block)
```

Comment:
`#method_missing` is much slower than straight delegation. Why change?

Reply:
The delegation is to a private method, and this throws off MASSIVE logs. This is a part of this change. If it's reimplemented as a delegation, then it will start to throw a LOT of messages about delegating to a private method.  

This approach keeps the messages from happening, and still raises on the non-defined methods. 

I pulled this approach from another gem that had the same problem... this approach was how they solved it.

Reply:
Let's just make the target methods public. See https://github.com/cequel/cequel/pull/378#discussion_r226803027